Allow to find system SOCI_* libraries

Add a second `find_library` call which will find the SOCI libraries in default locations if it hasn't been found already in `SOCI_LIBRARY_DIR`
See https://github.com/rstudio/rstudio/pull/11691

Author: Alexander Grund (TU Dresden)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index d2be417b40..6a42b4614d 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -467,12 +467,12 @@ endif()
 # find SOCI libraries
 if(UNIX)
    set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}/build/lib")
-   if(NOT APPLE AND RSTUDIO_USE_SYSTEM_SOCI)
-      set(SOCI_LIBRARY_DIR "/usr/lib")
-   endif()
    find_library(SOCI_CORE_LIB NAMES "libsoci_core.a" "soci_core" PATHS "${SOCI_LIBRARY_DIR}" NO_DEFAULT_PATH)
+   find_library(SOCI_CORE_LIB NAMES "libsoci_core.a" "soci_core")
    find_library(SOCI_SQLITE_LIB NAMES "libsoci_sqlite3.a" "soci_sqlite3" PATHS "${SOCI_LIBRARY_DIR}" NO_DEFAULT_PATH)
+   find_library(SOCI_SQLITE_LIB NAMES "libsoci_sqlite3.a" "soci_sqlite3")
    find_library(SOCI_POSTGRESQL_LIB NAMES "libsoci_postgresql.a" "soci_postgresql" PATHS "${SOCI_LIBRARY_DIR}" NO_DEFAULT_PATH)
+   find_library(SOCI_POSTGRESQL_LIB NAMES "libsoci_postgresql.a" "soci_postgresql")
    find_library(DL_LIB "dl")
    find_library(SQLITE_LIB "sqlite3")
    get_filename_component(SQLITE_LIB "${SQLITE_LIB}" REALPATH)
